src/providers/xref.provider.ts

```diff
@@ -30,15 +30,41 @@ function shouldProvide (context: Context): boolean {
 }
 
 async function getLabels (): Promise<string[]> {
-  const regex = /\\[\\[(\\w+)\\]\\]/g
-  const labels = await vscode.workspace.findFiles('**/*.adoc').then((files) =>
-    files
-      .map((uri) => readFileSync(uri.path).toString('utf-8'))
-      .join('\n')
-      .match(regex)
-      .map((result) => result.replace('[[', '').replace(']]', ''))
-  )
-  return labels
+  const files = await vscode.workspace.findFiles('**/*.adoc')
+  const contentOfFilesConcatenated = files
+    .map((uri) => readFileSync(uri.path).toString('utf-8'))
+    .join('\n')
+  const labelsFromLegacyBlock = await getLabelsFromLegacyBlock(contentOfFilesConcatenated)
+  const labelsFromShorthandNotation = await getLabelsFromShorthandNotation(contentOfFilesConcatenated)
+  const labelsFromLonghandNotation = await getLabelsFromLonghandNotation(contentOfFilesConcatenated)
+  return labelsFromLegacyBlock.concat(labelsFromShorthandNotation, labelsFromLonghandNotation)
+}
+
+async function getLabelsFromLonghandNotation (content: string): Promise<string[]> {
+  const regex = /\[id=(\w+)\]/g
+  const matched = content.match(regex)
+  if (matched) {
+    return matched.map((result) => result.replace('[id=', '').replace(']', ''))
```

Comment:
Might be better to use the (named) group? Since we are using `(\w+)` we can use the group reference `$1`

Reply:
seems we need a more recent version of TypeScript https://javascript.info/regexp-groups#searching-for-all-matches-with-groups-matchall

Reply:
`matchAll` is available on Node 12. Are you sure it does not work?

Reply:
I have this error:
![image](https://user-images.githubusercontent.com/1105127/204146709-bd571d28-202d-4a40-90f4-bec67cc9d796.png)

Reply:
You can safely update the target to `es2020`.

I'm planning to make the switch: https://github.com/asciidoctor/asciidoctor-vscode/pull/506

Reply:
I would prefer that we keep it in a separate Pull request. I'm not even sure how to change the target correctly neither what are the implications

Reply:
All good 👍🏻